Sources/Commands/Workspace.swift

```diff
@@ -480,8 +548,10 @@ public class Workspace {
             let specifier = RepositorySpecifier(url: externalManifest.url)
             let managedDependency = dependencyMap[specifier]!
 
-            // If we know the manifest is at a particular version, use that.
-            if let version = managedDependency.currentVersion {
+            if managedDependency.isEdited {
+                constraints.append(RepositoryPackageConstraint(container: specifier, versionRequirement: .any))
```

Comment:
This isn't going to do what you want it to do (we need a FIXME here). The solver will still merge in any of the other constraints and resolve using them. We should have test coverage for this (which would have showed it not working...)